draft-specifications/Model.md

```diff
@@ -526,69 +472,56 @@ local peer, the sent and received values will have the same type.
 
 ## Pass Invariant Equality
 
-A type holds equality invariant over passage between OCapN peers if values of that
-type can be passed from a local peer to a remote peer, then returned to the local peer,
-and the value that was sent is equal to the value received.
+OCapN defines a flavor of equality, nominally Equality.
+Two values may be Equal under Equality.
+Values of different OCapN types are not Equal.
+Values of the same OCapN type may be Equal under conditions specific to their
+shared type.
+All OCapN types maintain pass invariant Equality except Promise.
 
-> Equality does not guarantee a unique identity for all equivalent values.
+All OCapN implementations, regardless of language, must model all OCapN types
+such that Equality for the in-language representation of values is consistent
+with the Equality and Pass Invariance of Equality for their corresponding OCapN
+model type.
```

Comment:
Why? This seems both wrong as a concrete requirement and none of ocapn's business. It should be up to each language binding.

It is certainly wrong as stated, because it seems to refer to one of the language's existing notions of equality. For the endo binding, there is no JavaScript-level equality concept that corresponds to ocapn equality. Rather, the endo ocapn binding itself provides new equality predicates that, together, do correspond to ocapn equality. However, at least currently, we do not provide a single equality predicate that does so directly.

Reply:
I’m just going to remove for now, for tentative lack of verbiage that expresses my intent. I’m trying to express a constraint about a transitive round-trip invariant.

Reply:
I suspected that was what you were trying to get at. How about we talk about a language binding as a pair of functions, (naming? marshal/unmarshal, encode/decode, serialize/unserialize, f/g ?) where 
- f(passable-lanuage-value) -> ocapn-value
- g(ocapn-value) -> passable-language-value

and then state the round tripping requirement for the language side as 

for all ocapn-values `ov`, `ocapn-equal(ov, f(g(ov)))`

IOW, they round trip through the language-binding preserving ocapn equality.